include/arch/arm/aarch32/exc.h

```diff
@@ -73,25 +73,45 @@ GTEXT(z_arm_exc_exit);
 extern "C" {
 #endif
 
+struct __basic_sf {
+	sys_define_gpr_with_alias(a1, r0);
+	sys_define_gpr_with_alias(a2, r1);
+	sys_define_gpr_with_alias(a3, r2);
+	sys_define_gpr_with_alias(a4, r3);
+	sys_define_gpr_with_alias(ip, r12);
+	sys_define_gpr_with_alias(lr, r14);
+	sys_define_gpr_with_alias(pc, r15);
+	uint32_t xpsr;
+};
+
 struct __esf {
-	struct __basic_sf {
-		sys_define_gpr_with_alias(a1, r0);
-		sys_define_gpr_with_alias(a2, r1);
-		sys_define_gpr_with_alias(a3, r2);
-		sys_define_gpr_with_alias(a4, r3);
-		sys_define_gpr_with_alias(ip, r12);
-		sys_define_gpr_with_alias(lr, r14);
-		sys_define_gpr_with_alias(pc, r15);
-		uint32_t xpsr;
-	} basic;
+	struct __basic_sf basic;
+#if defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING)
+	float s[16];
+	uint32_t fpscr;
+	uint32_t undefined;
+#endif
+};
+
+struct __extended_esf {
+	/* Matches layout of struct __esf */
+	struct __basic_sf basic;
 #if defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING)
 	float s[16];
 	uint32_t fpscr;
 	uint32_t undefined;
 #endif
+#if defined(CONFIG_EXTRA_EXCEPTION_INFO)
+	/* Additional register state that is not stacked by
+	 * hardware on exception entry
+	 */
+	_callee_saved_t *callee;
+	uint32_t msp;
+	uint32_t exc_return;
+#endif
 };
 
-typedef struct __esf z_arch_esf_t;
+typedef struct __extended_esf z_arch_esf_t;
```

Comment:
So if my proposal is feasible (pls, investigate), this could be left as it is now. :)